clang/test/compiler-course/khovansky_d_lab1_test/khovansky_d_lab1_test.cpp

```diff
@@ -0,0 +1,71 @@
+// RUN: %clang_cc1 -load %llvmshlibdir/ImplicitConvPlugin_Khovansky_Dmitry_FIIT2_ClangAST%pluginext -plugin ImplicitConvPlugin -fsyntax-only %s 2>&1 | FileCheck %s
+
+// Проверка глобального скоупа
+// CHECK: In global scope:
+// CHECK-NEXT: int -> double: 1
+double global_par = 42;
+
+// Проверка задач варианта
+// CHECK: Function: sum
+// CHECK-NEXT: float -> double: 1
+// CHECK-NEXT: int -> float: 1
+
+// CHECK: Function: mul
+// CHECK-NEXT: double -> int: 1
+// CHECK-NEXT: float -> double: 1
+// CHECK-NEXT: float -> int: 1
+
+double sum(int a, float b) {
+    return a + b;
+}
+
+int mul(float a, float b) {
+    return a + sum(a, b);
+}
+
```

Comment:
Please, add some tests with implicit conversions of custom user created classes:

```
class X {
    X(int x);
}
```